client/network/src/service.rs

```diff
@@ -991,10 +991,17 @@ impl Metrics {
 	}
 }
 
-impl<B: BlockT + 'static, H: ExHashT> Future for NetworkWorker<B, H> {
-	type Output = Result<(), io::Error>;
+impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
+	/// Performs one action on the network, then returns.
```

Comment:
I am not quite sure I follow. If I read the logic below correctly this does not perform a single action, but block until the entire `NetworkWorker` is done.

Based on the pull request description I am guessing that this comment describes the behavior that we would like to have in the future. If so, would you mind mentioning that in the comment?